src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -378,65 +183,71 @@ public static void readItems(IItemHandlerModifiable handler, String tagName, NBT
         }
     }
 
-    public static boolean isBetweenInclusive(long start, long end, long value) {
-        return start <= value && value <= end;
+    /**
+     * @param array Array sorted with natural order
+     * @param value Value to search for
+     * @return Index of the nearest value lesser or equal than {@code value},
+     * or {@code -1} if there's no entry matching the condition
+     */
+    public static int nearestLesserOrEqual(@Nonnull long[] array, long value) {
+        int low = 0, high = array.length - 1;
+        while (true) {
+            int median = (low + high) / 2;
+            if (array[median] <= value) {
+                if (low == high) return low;
+                low = median + 1;
+            } else {
+                if (low == high) return low - 1;
+                high = median - 1;
+            }
+        }
     }
 
     /**
-     * Capitalizes string, making first letter upper case
-     *
-     * @return capitalized string
+     * @param array Array sorted with natural order
+     * @param value Value to search for
+     * @return Index of the nearest value lesser than {@code value},
+     * or {@code -1} if there's no entry matching the condition
      */
-    public static String capitalizeString(String string) {
-        if (string != null && string.length() > 0)
-            return string.substring(0, 1).toUpperCase() + string.substring(1);
-        return "";
+    public static int nearestLesser(@Nonnull long[] array, long value) {
+        int low = 0, high = array.length - 1;
+        while (true) {
+            int median = (low + high) / 2;
+            if (array[median] < value) {
+                if (low == high) return low;
+                low = median + 1;
+            } else {
+                if (low == high) return low - 1;
+                high = median - 1;
+            }
+        }
     }
 
     /**
-     * @return lowest tier that can handle passed voltage
+     * @return Lowest tier of the voltage that can handle {@code voltage}; that is,
+     * a voltage with value greater than equal than {@code voltage}. If there's no
+     * tier that can handle it, {@code MAX} is returned.
      */
     public static byte getTierByVoltage(long voltage) {
-        if (voltage > V[GTValues.MAX]) return GTValues.MAX;
-        return tierByVoltage.ceilingEntry(voltage).getValue();
+        return (byte) Math.min(GTValues.MAX, nearestLesser(V, voltage) + 1);
```

Comment:
How does this new method perform compared to the previous? This method is called a lot, and with older implementations caused hotspots.

Reply:
well it doesn't use treemap and boxed primitive anymore so I assume it's at least marginally faster? Never profiled it, though.